sycl/doc/extensions/proposed/sycl_ext_oneapi_group_load_store_extended.asciidoc

```diff
@@ -0,0 +1,299 @@
+= sycl_ext_oneapi_group_load_store
+
+:source-highlighter: coderay
+:coderay-linenums-mode: table
+
+// This section needs to be after the document title.
+:doctype: book
+:toc2:
+:toc: left
+:encoding: utf-8
+:lang: en
+:dpcpp: pass:[DPC++]
+:language: {basebackend@docbook:c++:cpp}
+
+== Introduction
+
+IMPORTANT: This specification is a draft.
+
+
+== Notice
+
+[%hardbreaks]
+Copyright (c) 2022 Intel Corporation.  All rights reserved.
+
+Khronos(R) is a registered trademark and SYCL(TM) and SPIR(TM) are trademarks
+of The Khronos Group Inc.  OpenCL(TM) is a trademark of Apple Inc. used by
+permission by Khronos.
+
+== Dependencies
+
+This extension is written against the SYCL 2020 specification, Revision 6.
+All references below to the "core SYCL specification" or to section numbers in
+the SYCL specification refer to that revision.
+
+This extension also depends on the following other SYCL extensions:
+
+* link:https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/proposed/sycl_ext_oneapi_group_sort.asciidoc[
+  sycl_ext_oneapi_group_sort] (`group_with_scratchpad` class used as GroupHelper).
+
+* link:../experimental/sycl_ext_oneapi_properties.asciidoc[
+  sycl_ext_oneapi_properties].
+
+== Overview
+
+This extension defines free functions for load/store operations within work
+group scope.
+
+== Single Value Functions
+
+[source,c++]
+----
+namespace sycl::ext::oneapi::experimental {
+
+// Load API scalar
+template<typename Group, typename InputIteratorT,
+         typename OutputT, typename Properties = properties<>>
+void group_load(Group g, InputIteratorT in_ptr,
+                      OutputT& out, Properties = {}); // (1)
+
+// Store API scalar
+template<typename Group, typename InputT,
+         typename OutputIteratorT, typename Properties = properties<>>
+void group_store(Group g, const InputT& in,
+                      OutputIteratorT out_ptr, Properties = {}); // (2)
+
+}
+----
+
+(1) _Preconditions_: `in_ptr` must be the same for all work-items
+in the group.
+
+_Constraints_: Only available if `Group` is a work-group or sub-group.
+
+_Effects_: Loads single element from `in_ptr` to `out` by using the `g` group
+object to identify memory location. For the 1d case it would be:
+`g.get_group_id(0) * g.get_local_range(0)  + g.get_local_id(0)`
```

Comment:
Can we specify how exactly group `g` object identifies memory location which needs to be loaded? Based on the example, I _assume_ that every work item loads an element address of which is calculated as `in_ptr + g.get_local_linear_id()` - is it correct understanding?

It is just I don't think that it is enough for the spec to provide an example instead of actually documenting how it is supposed to work.

Reply:
yes, you are right, updated description everywhere